### PR TITLE
Fix protocol compatibility check

### PR DESCRIPTION
**Pull request by Henningstone**
This completes the protocol compatibility check system.
- Added the 0.6 netversion to `modapi/compatibility.h`
- `CServer::GetClientProtocolCompatibility` checks now for all protocols, not only for 0.7ModAPI
- Little code cleanups

### CODE REVIEW

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -1915,5 +1902,5 @@ void CServer::SendInitialData(int ClientID)
 
 bool CServer::GetClientProtocolCompatibility(int ClientID, int Protocol) const
 {
-	return (m_aClients[ClientID].m_Protocol == MODAPI_CLIENTPROTOCOL_TW07MODAPI);
+	return (m_aClients[ClientID].m_Protocol == Protocol);
```
Comment (necropotame) [inline src/engine/server/server.cpp]: Remove this, the old version is correct. This function check if the protocol is compatible with GUI, Sprites, ...

Comment (Henningstone) [inline src/engine/server/server.cpp]: But then, the function would really not match it's purpose :open_mouth: And the `int Protocol` argument will be completely useless?
I thought this is the purpose of the function no?
`GetClientProtocolCompatibility(1, MODAPI_CLIENTPROTOCOL_TW07MODAPI);` would check if the protocol of client 1 is compatible with GUI, Sprites, ...

Comment (necropotame) [inline src/engine/server/server.cpp]: It's useless now, but if we have several version of ModAPI, the parameter will he usefull. Check message.cpp, I use it.

Comment (Henningstone) [inline src/engine/server/server.cpp]: Oh well, then I misunderstood the purpose of the function. I'll change it back.
Should there be a other function which does what I thought it should do?

**Pull request merged**